subsys/fs/shell.c

```diff
@@ -36,12 +36,21 @@ static struct fs_mount_t fatfs_mnt = {
 
 /* TODO: Implement dynamic storage dev selection */
 #ifdef CONFIG_FS_LITTLEFS_BLK_DEV
-FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(lfs_data);
+
+#if defined(CONFIG_DISK_DRIVER_SDMMC)
+#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
+#elif IS_ENABLED(CONFIG_DISK_DRIVER_MMC)
```

Comment:
```
#if defined(.....
....
#elif IS_ENABLED(...
```
I think IS_ENABLED macro should be used for both cases. Just to be consistent.

Reply:
it is a common approach:
```
samples/subsys/fs/littlefs/src/main.c:313:#if defined(CONFIG_DISK_DRIVER_SDMMC)
samples/subsys/fs/littlefs/src/main.c-314-#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
samples/subsys/fs/littlefs/src/main.c-315-#elif IS_ENABLED(CONFIG_DISK_DRIVER_MMC)
samples/subsys/fs/littlefs/src/main.c-316-#define DISK_NAME CONFIG_MMC_VOLUME_NAME
samples/subsys/fs/littlefs/src/main.c-317-#else
samples/subsys/fs/littlefs/src/main.c-318-#error "No disk device defined, is your board supported?"
--
tests/drivers/disk/disk_access/src/main.c:17:#if defined(CONFIG_DISK_DRIVER_SDMMC)
tests/drivers/disk/disk_access/src/main.c-18-#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
tests/drivers/disk/disk_access/src/main.c-19-#elif IS_ENABLED(CONFIG_DISK_DRIVER_MMC)
tests/drivers/disk/disk_access/src/main.c-20-#define DISK_NAME CONFIG_MMC_VOLUME_NAME
tests/drivers/disk/disk_access/src/main.c-21-#elif IS_ENABLED(CONFIG_DISK_DRIVER_RAM)
tests/drivers/disk/disk_access/src/main.c-22-#define DISK_NAME CONFIG_DISK_RAM_VOLUME_NAME
--
tests/drivers/disk/disk_performance/src/main.c:15:#if defined(CONFIG_DISK_DRIVER_SDMMC)
tests/drivers/disk/disk_performance/src/main.c-16-#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
tests/drivers/disk/disk_performance/src/main.c-17-#elif IS_ENABLED(CONFIG_DISK_DRIVER_MMC)
tests/drivers/disk/disk_performance/src/main.c-18-#define DISK_NAME CONFIG_MMC_VOLUME_NAME
tests/drivers/disk/disk_performance/src/main.c-19-#elif IS_ENABLED(CONFIG_DISK_DRIVER_RAM)
tests/drivers/disk/disk_performance/src/main.c-20-#define DISK_NAME CONFIG_DISK_RAM_VOLUME_NAME
```

Reply:
..... Or was written this way once and the copied to all places. 
Anyway it works botch ways, but looks weird and produces a question is there some hidden meaning in using different checks in #if and #elif...
I'm okay with current approach. It's up to you whether change this or not.

Reply:
changed to `#if defined`

Reply:
+1